### ISSUE TITLE
Payment percentage fee.php file

### ISSUE THREAD

**Issue by leonhelmus** [Adyen/magento#596]
lin 59 and 60

`$grandTotal = $address->getGrandTotal() - ($paymentMethodFee + $paymentMethodFeeTax);
        $fee = ($address->getGrandTotal() / 100) * $percentageFee;`

should probably be 

`$grandTotal = $address->getGrandTotal() - ($paymentMethodFee + $paymentMethodFeeTax);
        $fee = ($grandTotal / 100) * $percentageFee;`

**Comment by leonhelmus** [comment]
> class Adyen_Fee_Model_Total_PaymentFee_Invoice extends Mage_Sales_Model_Order_Invoice_Total_Abstract
> {
>     public function collect(Mage_Sales_Model_Order_Invoice $invoice)
>     {
>         $invoice->setGrandTotal($invoice->getGrandTotal()+$invoice->getPaymentFeeAmount());
>         $invoice->setBaseGrandTotal($invoice->getBaseGrandTotal()+$invoice->getBasePaymentFeeAmount());
>         return $this;
>     }
> }

**Comment by leonhelmus** [comment]
if i want to make invoices i will have to to add taxes to the grandtotal otherwise total due will not be zero

**Comment by rikterbeek** [comment]
Hi @leonhelmus,

I will look into your code changes. You can always do a pull request to our develop branch if you want some code changes in our code.

Regards,
Rik
Adyen

**Comment by rikterbeek** [comment]
@leonhelmus,

Your first comment is correct I will change this. The second comment is the same code as we have in the module.

Regards,
Rik
Adyen

**Comment by rikterbeek** [comment]
Hi @leonhelmus,

I checked it and we are doing a percentage over the grand total so the percentage has the included tax as well. So tax is raised to the percentage but not stated in a separate tax class is this what you want ?

Regards,
Rik
Adyen

**Comment by leonhelmus** [comment]
yes it is but allready fixed that myself (cant push it today probably tomorrow if i have time). aswell for mailing and cutomer account in new order invoice and creditmemo. Now i only need a fix for refunding

**Comment by rikterbeek** [comment]
@leonhelmus,

If you could show me the code changes then I can validate if and how we can merge this into our module. Please let me know where I can find the code changes that you have made. The refunded part is done for payment fee and you can see the code changes in issue #593 .

Regards,
Rik
Adyen

**Comment by leonhelmus** [comment]
here beneath you have my git repository if you have any questions please ask and i'll respond. This code has worked using our magento version 1.9.2.

https://github.com/leonhelmus/magento

**Comment by seansan** [comment]
@leonhelmus Is this still a problem?
